RELEASING.md

```diff
@@ -0,0 +1,24 @@
+# Releasing new version
+
+## Step-by-step
+
+In order to release a new version, there are several steps that must happen:
+
+ 0. make sure all issues (in GitHub and in JIRA) targeted to the version are resolved
+ 1. create a pull request in which you change the [version string](https://github.com/modcluster/mod_proxy_cluster/blob/main/native/include/mod_proxy_cluster.h#L21) according to the format described in [another section](#version-naming-conventions)
```

Comment:
We need to define here what will the snapshot versions be called.

Reply:
I know, I'd like to use the hash here. I'll see what can we do here.

Reply:
Adding dynamic git commit hash is really aspect of the build scripts, that infer the git head hash and use it during compilation in the version somewhere.

All we need to do here within the context of this document, is to figure out, what we put in the version string in between actual tags.

Reply:
> Adding dynamic git commit hash is really aspect of the build scripts, that infer the git head hash and use it during compilation in the version somewhere.
>
> All we need to do here within the context of this document, is to figure out, what we put in the version string in between actual tags.

That would need git, which we don't have to have. What I was thinking about was something like pre/commit script that would do it automatically. But that would be noisy, imho (maybe post PR? I haven't explored it yet).

Reply:
Maybe we might keep it simple and use just something like `major.minor.micro.Dev` with an optional number at the end. We would increment it once in a while if we make some major changes...

Reply:
> > Adding dynamic git commit hash is really aspect of the build scripts, that infer the git head hash and use it during compilation in the version somewhere.
> > All we need to do here within the context of this document, is to figure out, what we put in the version string in between actual tags.
> 
> That would need git, which we don't have to have. What I was thinking about was something like pre/commit script that would do it automatically. But that would be noisy, imho (maybe post PR? I haven't explored it yet).

Yeah, that can be done with GH Action too, but way too noisy and hacky.

Reply:
> Maybe we might keep it simple and use just something like `major.minor.micro.Dev` with an optional number at the end. We would increment it once in a while if we make some major changes...
> 
> For example after we release `2.0.0.Final`, the next version string would be `2.0.1.Dev`. If we make only small changes, we can do `2.0.1.Dev1`, `2.0.1.Dev2`.... and if we make some bigger ones, we can do `2.1.0.Dev`. Or we can have only Dev and increment only the micro if we don't insist on strict continuity.

I guess it doesn't really matter,  I am mostly concerned about protocol negotiation, but that will be handled by parsing the 'MCMP protocol version' (which is currently reported as `0.2.1`). If the version string itself is to be parsed out, we really care for the Final releases. So, anything is fine really.

Reply:
In httpd they have `STATUS` file and a header macros. You can see here how does release of 2.4.61 look like

https://github.com/apache/httpd/commit/d49b8e01d885f85bdcec355190fbcf066d0afc50

version got increased to 2.4.62 with "In Development" status.

Reply:
@jfclere @rhusar I've updated the RELEASING.md following the discussion.

Reply:
👍🏼